crates/eww/src/widgets/widget_definitions.rs

```diff
@@ -527,6 +527,42 @@ fn parse_icon_size(o: &str) -> Result<gtk::IconSize> {
     }
 }
 
+fn svg_to_pixbuf(
+    path: std::path::PathBuf,
+    image_width: i32,
+    image_height: i32,
+    fill: &str,
+) -> Result<Option<gtk::gdk_pixbuf::Pixbuf>> {
```

Comment:
this api seems a bit weird
why return a `Result<Option<_>>` here? if the operation fails, just return an error

Reply:
I did it that way to avoid using unwrap :laughing:, and also because the pixbuf() method of PixbufLoader can return None if the image is empty.

Reply:
i think it'd be better to move that away from this function then and have it only handle the svg manipulation then
maybe even consider inlining this function entirely

Reply:
i think it'd be better to move that away from this function then and have it only handle the svg manipulation then
maybe even consider inlining this function entirely

Reply:
I didn't understand, you mean to move the svg_to_pixbuf functionality inside the image widget block where the SVG is handled. Sorry for my bad English.

Reply:
yes